MyToolsYourToolsBackend/MyToolsYourToolsBackend.Application/Services/RentService.cs

```diff
@@ -30,35 +30,41 @@ public bool CheckIfUserHasEnoughPoints(Guid userId)
             return new PointsModificationToolBorrowingStrategy().Modify(user.Points) > 0;
         }
 
-        public Rent AddRent(RentForCreationDto rent)
+        public Rent AddRent(RentDto rent)
         {
             var rentToSave = Mapper.Map<Rent>(rent);
             var offer = _dbContext.Offers.FirstOrDefault(o => o.Id == rent.OfferId);
             var borrower = _dbContext.Users.FirstOrDefault(u => u.Id == rent.BorrowerId);
             var rentingUser = _dbContext.Users.FirstOrDefault(r => r.Id == offer.OwnerId);
 
             borrower.Rents.Add(rentToSave);
-            offer.Status = Domain.Enums.OfferStatus.Rented;
+            offer.Status = OfferStatus.Rented;
 
             // delete all remaining rentRequests of this offer
-            var rentRequestsToRemove = _dbContext.Notifications
+            IList<Notification> rentRequestsToRemove = _dbContext.Notifications
                 .Where(n => n.OfferId == rent.OfferId
                         && n.Type == NotificationType.RentRequest
-                        && n.TargetUserId != borrower.Id);
-            _dbContext.RemoveRange(rentRequestsToRemove);
+                        && n.TargetUserId != borrower.Id).ToList<Notification>();
 
             _pointsService.ModifyPoints(rentingUser, new PointsModificationToolRentingStrategy());
-
             _pointsService.ModifyPoints(borrower, new PointsModificationToolBorrowingStrategy());
 
+
             if (_dbContext.SaveChanges() == 0)
             {
                 throw new Exception("Could not add rent");
-            }            
+            }
+
+            foreach (var rentRequest in rentRequestsToRemove)
+            {
```

Comment:
RemoveRange zamiast tego ?

Reply:
oj nie, to leci przez serwis Michał, żeby kaucje zwrócił tym, którzy nie otrzymują oferty

Reply:
a ok.:P